packages/realm-react/src/UserProvider.tsx

```diff
@@ -49,9 +49,10 @@ export const UserProvider: React.FC<UserProviderProps> = ({ fallback: Fallback,
 
   useEffect(() => {
     const event = () => {
-      if (app.currentUser?.id != user?.id) {
-        setUser(app.currentUser);
-      }
+      console.log("UserProvider: user changed");
+      console.log("UserProvider: app.currentUser", app.currentUser);
+      console.log("UserProvider: app.currentUser.id", app.currentUser);
+      setUser(app.currentUser);
```

Comment:
🙂

Reply:
Somehow I lost the change I actually wanted to commit.  This line is now
```
setUser( app.currentUser ? new Proxy(app.currentUser, {}) : null );
```

Reply:
Is it worth adding a test for this btw?

Reply:
It's worth it, but I don't have time, but I'm not actually scheduled to work on this right now.  I just want things to work.  
The UserProvider needs tests, at the time of creation it wasn't clear how to deploy an app for functional testing, but now we can do this.

Reply:
https://github.com/realm/realm-js/issues/6212. For the next phase of `realm react` we have planned

Reply:
Great!

Reply:
FYI, looks like the `new Proxy()` implementation was never committed.

Reply:
🤯

Reply:
Must be a rebase issue.  Guess that's why we should write tests :D